spd_conn.cc

```diff
@@ -255,6 +274,17 @@ void spider_free_conn_from_trx(
             pthread_mutex_unlock(&spider_conn_mutex);
             spider_free_conn(conn);
           } else {
+            long mutex_num=0;
+            if(ip_port_conn = (SPIDER_IP_PORT_CONN*) my_hash_search_using_hash_value(&spider_ipport_conns, conn->conn_key_hash_value, (uchar*) conn->conn_key, conn->conn_key_length))
+            {/* exists */
+              if(spider_param_max_connections())
```

Comment:
after spider_get_conn_from_idle_connection return NULL (may be timeout),
if(conn==NULL && spider_max_connections ==0)
then, create conn ?
do you approve this behavior ?  I this way, some threads cost 20s seconds more.

In other way, we use a thd variables to keep spider_param_max_connections.  It must involve some changes about mysql server layer.